lib/gprocess.c

```diff
@@ -1446,13 +1446,25 @@ void
 g_process_finish(void)
 {
 #ifdef SYSLOG_NG_HAVE_ENVIRON
-  for (gint i = 0; i < env_ctr && environ[i] != NULL; i++)
-    g_free(environ[i]);
-
-  g_free(environ);
-
-  if (process_opts.argv_orig)
-    free(process_opts.argv_orig);
+  /**
+   * There were various attempts to eliminate this leak. The reason
```

Comment:
The _"to eliminate this leak"_ expression is not self-explaining without the context of this patch.

I would rather use the _"to free  the `environ` variable"_ expression.